File: LENTABP10/src/main/java/com/lenta/lentabp10/models/task/TaskSaveModel.java
package com.lenta.lentabp10.models.task;

import com.lenta.lentabp10.models.repositories.ITaskRepository;
import com.lenta.shared.models.core.ProductInfo;

import java.util.List;

public class TaskSaveModel {

    private TaskDescription taskDescription;
    private ITaskRepository taskRepository;

    public TaskSaveModel(TaskDescription taskDescription, ITaskRepository taskRepository) {
        this.taskDescription = taskDescription;
        this.taskRepository = taskRepository;
    }

    public String getPerNo(){
        return taskDescription.getPerNo();
    }

    public String getPrinter(){
        return taskDescription.getPrinter();
    }

    public String getTaskName(){
        return taskDescription.getTaskName();
    }

    public TaskType getTaskType(){
        return taskDescription.getTaskType();
    }

    public String getTkNumber(){
        return taskDescription.getTkNumber();
    }

    public String getStorloc(){
        return taskDescription.getStock();
    }

    public String getIpAdress(){
        return taskDescription.getIpAdress();
    }

    public List<String> getMoveTypes(){
        return taskDescription.getMoveTypes();
    }

    public List<String> getGisControls(){
        return taskDescription.getGisControls();
    }

    public List<String> getMaterialTypes(){
        return taskDescription.getMaterialTypes();
    }

    public List<ProductInfo> getMaterials(){
        return taskRepository.getProducts().getProducts();
    }

    public List<TaskWriteOffReason> getWriteOffReasons(){
        return taskRepository.getWriteOffReasons().getWriteOffReasons();
    }

    public List<TaskExciseStamp> getExciseStamps(){
        return taskRepository.getExciseStamps().getExciseStamps();
    }

}

File: writeoff_breakage/src/main/java/com/lenta/bp10/models/task/TaskSaveModel.kt
package com.lenta.bp10.models.task

import com.lenta.bp10.models.repositories.ITaskRepository
import com.lenta.shared.models.core.ProductInfo

class TaskSaveModel(val taskDescription: TaskDescription, val taskRepository: ITaskRepository) {

    fun getPerNo(): String {
        return taskDescription.perNo
    }

    fun getPrinter(): String {
        return taskDescription.printer
    }

    fun getTaskName(): String {
        return taskDescription.taskName
    }

    fun getTaskType(): TaskType {
        return taskDescription.taskType
    }

    fun getTkNumber(): String {
        return taskDescription.tkNumber
    }

    fun getStorloc(): String {
        return taskDescription.stock
    }

    fun getIpAdress(): String {
        return taskDescription.ipAddress
    }

    fun getMoveTypes(): List<WriteOffReason> {
        return taskDescription.moveTypes
    }

    fun getGisControls(): List<String> {
        return taskDescription.gisControls
    }

    fun getMaterialTypes(): List<String> {
        return taskDescription.materialTypes
    }

    fun getMaterials(): List<ProductInfo> {
        return taskRepository.getProducts().getProducts()
    }

    fun getWriteOffReasons(): List<TaskWriteOffReason> {
        return taskRepository.getWriteOffReasons().getWriteOffReasons()
    }

    fun getExciseStamps(): List<TaskExciseStamp> {
        return taskRepository.getExciseStamps().getExciseStamps()
    }
}